src/box/memtx_tree.cc

```diff
@@ -289,25 +322,20 @@ tree_iterator_next_base(struct iterator *iterator, struct tuple **ret)
 	} else {
 		memtx_tree_iterator_next(&index->tree, &it->tree_iterator);
 	}
-	tuple_unref(it->current.tuple);
 	struct memtx_tree_data<USE_HINT> *res =
 		memtx_tree_iterator_get_elem(&index->tree, &it->tree_iterator);
-	if (res == NULL) {
-		iterator->next = tree_iterator_dummie;
-		it->current.tuple = NULL;
-		*ret = NULL;
-	} else {
-		*ret = res->tuple;
-		tuple_ref(*ret);
-		it->current = *res;
-	}
 	struct index *idx = iterator->index;
 	struct space *space = space_by_id(iterator->space_id);
 	/*
 	 * Pass no key because any write to the gap between that
 	 * two tuples must lead to conflict.
 	 */
-	memtx_tx_track_gap(in_txn(), space, idx, *ret, ITER_GE, NULL, 0);
+	memtx_tx_track_gap(in_txn(), space, idx,
```

Comment:
Pleas don't move the code that handles `currect` and `ret` below memtx_tx_track_gap. It actually can do something with tree and tree iterator would be broken.